docker/Dockerfile.ubuntu22.04

```diff
@@ -1,5 +1,5 @@
-FROM nvcr.io/nvidia/cuda:12.4.1-base-ubuntu22.04 AS builder
-ARG GOLANG_VERSION=1.21.5
+FROM nvcr.io/nvidia/cuda:12.5.1-base-ubuntu22.04 AS builder
+ARG GOLANG_VERSION=1.22.4
```

Comment:
Do we need to update Golang to 1.22.5 here as well, or I guess the Makefile overrides it.

Reply:
Discussed offline, marking resolved.